### PR TITLE
fix: Bad date type in email text report for table chart

### PR DESCRIPTION
**Pull request by codemaster08240328**
### SUMMARY

When creating a report that sends a table chart with dates, and choose the option send by email as text, the dates show badly formatted

It is because `/api/v1/chart/data` endpoint returns datetime64 data which represents as milliseconds.

Need to convert is as date type when we create pandas Dataframe from the result data.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

BEFORE:
image

AFTER: 
image

### TESTING INSTRUCTIONS

1. Create a table chart which contains date.
2. Create a text email report.
3. Check your email.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/utils/csv.py [hunk 1]
```diff
@@ -97,11 +99,18 @@ def get_chart_dataframe(
...
     pd.set_option("display.float_format", lambda x: str(x))
     df = pd.DataFrame.from_dict(result["result"][0]["data"])
 
+    # if any column type is equal to 2, need to convert data into datetime timestamp for that column.
+    if GenericDataType.TEMPORAL in result["result"][0]["coltypes"]:
+        for i in range(len(result["result"][0]["coltypes"])):
+            if result["result"][0]["coltypes"][i] == GenericDataType.TEMPORAL:
+                df[result["result"][0]["colnames"][i]] = df[
+                    result["result"][0]["colnames"][i]
+                ].astype("datetime64[ms]")
```
Comment (diegomedina248) [inline superset/utils/csv.py:112]: will this work for all databases?

**Comment by rusackas** [comment]
@eschutho do you have a convenient means of testing this PR? Not sure how to test on the ephemeral envs.

**Review by AAfghahi — COMMENTED** [review]
Looks good, do you mind writing a test where it converts the datetime?

**Comment by eschutho** [comment]
@rusackas unfortunately, no, we can't test reports on ephemeral but I can pull it down locally and test if that helps.

**Review by rusackas — APPROVED** [review]
LGTM!

**Pull request merged**